src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -167,21 +167,25 @@ private static List<MethodCategory> GetMethodCategories(Compilation compilation)
             var methodCategories = new List<MethodCategory> {
                 new MethodCategory(IsPropertyGetter, true,
                     PropertyGetterRule,
-                    WellKnownTypes.InvalidOperationException(compilation), WellKnownTypes.NotSupportedException(compilation)),
+                    compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemInvalidOperationException),
```

Comment:
I see that we use `GetOrCreateTypeByMetadataName` at some places and `GetTypeByMetadataName` at others. Is there a reasoning behind how to chose one over another at these places?

Reply:
All calls to the `WellKnownTypes` and `WellKnownTypeProvider` have been converted to `GetOrCreateTypeByMetadataName` while all previous calls to `GetTypeByMetadataName` were kept as they were. 

I personally don't see any "good" reason to keep calls to `GetTypeByMetadataName` and I am fine to add one more commit taking care of that if that's aligned with your way of seeing the code after this PR.